primitive-types/src/lib.rs

```diff
@@ -110,6 +110,44 @@ impl U256 {
 	pub fn full_mul(self, other: U256) -> U512 {
 		U512(uint_full_mul_reg!(U256, 4, self, other))
 	}
+
+	/// Lossy saturating conversion from a `f64` to a `U256`.
+	///
+	/// The conversion follows roughly the same rules as converting `f64` to other
```

Comment:
"roughly"? If there are any differences they should be documented. I'd prefer there were none though.

I'd also like to see some docs/examples about how `f64`s are truncated, e.g.`10.5_f64` –> `10_u256`.

Reply:
Yes, cool idea. Would it suffice to have a unit test like so:

```rs
#[test]
fn f64_to_u256_truncation_examples() {
	assert_eq!(U256::from_f64_lossy(10.5), 10.into());
}
```

with some examples? Do you have any other examples in mind? I ask because most of these corner cases are already covered in the tests.

Reply:
I agree that the unit tests seem to cover it all, I was referring to the docs. If you want to also provide examples that's even better.

So yeah, avoid vague language like "roughly" (or expand on it) and explain to users how their floats are cast, "The conversion behaves like [`trunc`](https://doc.rust-lang.org/std/primitive.f64.html#method.trunc)." or something along those lines?